.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,39 @@
+name: static-analysis
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  static_analysis:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php:
+          - '7.4'
+          - '8.0'
+          - '8.1'
+          - '8.2'
+          - '8.3'
+          - '8.4'
```

Comment:
Run only on a single PHP version, please. No need to create matrix at all.

Since it's a static analysis, then library code isn't executed. The only thing that is executed is PhpStan, and we don't need to test how it performs on every PHP version.